src/translation-api.js

```diff
@@ -0,0 +1,80 @@
+import { and, eq } from 'drizzle-orm'
+import { kCreateWithDocId, kSelect } from './datatype/index.js'
+import { hashObject } from './utils.js'
+
+export default class TranslationApi {
+  /** @type {Map<String,Set<import('@mapeo/schema/dist/types.js').SchemaName>>} */
+  #translatedLanguageCodeToSchemaNames = new Map()
+
+  /**
+   * @param {Object} opts
+   * @param {import('./datatype/index.js').DataType<
+   *   import('./datastore/index.js').DataStore<'config'>,
+   *   typeof import('./schema/project.js').translationTable,
+   *   'translation',
+   *   import('@mapeo/schema').Translation,
+   *   import('@mapeo/schema').TranslationValue
+   * >}  opts.dataType
+   * @param {typeof import('./schema/project.js').translationTable} opts.table
+   */
+  constructor({ dataType, table }) {
+    this.dataType = dataType
+    this.table = table
+  }
+
+  /**
+   * @param {import('@mapeo/schema').TranslationValue} value
+   */
+  async put(value) {
+    // const { message, ...identifiers } = value
+    const docId = hashObject(value)
+    let existing
+    try {
+      existing = await this.dataType.getByDocId(docId)
+    } catch (e) {
+      existing = await this.dataType[kCreateWithDocId](docId, value)
+    }
+    if (existing) {
```

Comment:
`existing` will always be truthy here, maybe this should be in the `try` block?

Reply:
oops, yeah, you're right